docs/state.md

```diff
@@ -1,35 +1,26 @@
-- AMD GPUs: Changing resolution, using DRI_PRIME and doing various other things can cause crashes, but `echo high | sudo tee /sys/bus/pci/drivers/amdgpu/0000:??:??.?/power_dpm_force_performance_level` or adding `amdgpu.dpm=0` to the kernel commandline stops these crashes.
-- MacPro7,1: Users have encountered PCIE Address Space issues, with auto remap breaking.
-
-## Not working
-
-- Custom graphics on Touchbar: There is currently no Linux driver for the Touchbar's "Touch Bar Display" device configuration, which is what macOS uses, and gives full control over the display to the Operating System.
-- T2 Secure Enclave Processor (Touch ID, storing encryption keys on the T2)
-- The T2's onboard Audio Video Encoder (used for Sidecar on macOS)
-- Graphics switching without rebooting (gmux)
-
-## Other
-
+|Feature|Status|Upstream|Notes|Issues/links|
+|-|-|-|-|-|
+|Internal Drive / SSD|🟢 Working|🟢 Kernel 5.4||[Filesystem notes](#filesystem-notes)|
+|Screen|🟢 Working|🟢 Yes|||
+|USB|🟢 Working|🟢 Yes|||
+|Keyboard|🟢 Working|🔴 No||[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Camera|🟢 Working|🟢 Yes|||
+|Wi-Fi|🟢 Working|🔴 No|Requires macOS firmware|[Setup guide](https://wiki.t2linux.org/guides/wifi-bluetooth/)|
+|Bluetooth|🟢 Working|🔴 No|Requires macOS firmware only for devices with BCM4377 chip. Also, Bluetooth glitches on devices with BCM4377 Chip if connected to a 2.4 Ghz Wi-Fi connection. Thus, in order to use Bluetooth either turn off your Wi-Fi or use a 5Ghz Wi-Fi connection.|[Setup guide](https://wiki.t2linux.org/guides/wifi-bluetooth/)|
+|Touch Bar|🟢 Working|🔴 No|There is support for the "Touch Bar Keyboard" device configuration, where only the Function Keys or the Media/Brightness Control Keys can be shown. There is work in progress support for custom graphics on the Touch Bar.|[tiny-dfr](https://github.com/kekrby/tiny-dfr)<br>[apple-ib](https://github.com/t2linux/apple-ib-drv)|
+|Suspend|🟡 Partially working|🟢 Yes|A firmware upgrade attached to macOS Sonoma broke suspend. Some users were having difficulty with it even before Sonoma.|[#53](https://github.com/t2linux/T2-Ubuntu-Kernel/issues/53)|
+|Trackpad|🟡 Partially working|🔴 No|Functions, but it is far from the experience on macOS (No force touch or palm rejection). Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered.|[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Audio|🟡 Partially working|🔴 No|With proper configuration audio can work, however it is not stable in some older kernels and switching between speakers and when using the microphone. Microphone volume is low in some Macs.|[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Hybrid Graphics|🟡 Partially working|🟡 Partial|If the device has a dedicated AMD GPU (15 and 16 inch MacBookPro's) as well as an Intel iGPU, the iGPU can be used, but this breaks resume.|[Hybrid Graphics](https://wiki.t2linux.org/guides/hybrid-graphics/)|
```

Comment:
using igpu no longer breaks resume so you can remove that bit. maybe also mention that toggling the dgpu power doesn't work